### PR TITLE
Adds the conveyor belt placer

### PR DESCRIPTION
**Pull request by Citinited**
This PR adds the conveyor belt placer (CBP), which is a glorified bag for conveyor belts. The purpose of this PR is to make conveyor belt placement much quicker and easier (currently it's rather time-consuming to make even short belts).

With the CBP you can:
- Click on the floor with the CBP to apply a belt there, assuming there are belts inside the CBP.
- Click on the CBP with a lever to link all belts with the lever.
- Click on loose belts with the CBP to pick them up.

The normal CBP can be obtained at any autolathe, and has a capacity of 7 belts. The bluespace CBP has to be researched before you can obtain it, and has a capacity of 50 belts.

**Obligatory picture!**

image

 The bluespace one is animated, the normal one is not.
 
🆑 
add: Adds the conveyor belt placer and bluespace conveyor belt placer, allowing you to much more easily create conveyor belts. The former can be gotten at any autolathe, the latter must be researched first.
add: Use a conveyor belt lever on aforementioned item to link all belts inside the placer with that lever.
/🆑

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
That sprite doesn't look very good.

The idea is interesting though.

File: code/modules/recycling/belt-placer.dm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	can_hold = list(/obj/item/conveyor_construct)
+	max_w_class = WEIGHT_CLASS_BULKY
+	max_combined_w_class = 28 //7 belts
+	allow_quick_gather = TRUE
+	allow_quick_empty = TRUE
+	display_contents_with_number = TRUE
+	use_to_pickup = TRUE
+	origin_tech = "engineering=1"
+
+/obj/item/storage/conveyor/bluespace
```
Comment (Fox-McCloud) [inline code/modules/recycling/belt-placer.dm]: I'd rather not have more infinite storage items, please.

The ore is bag is bad enough.

Comment (Citinited) [reply code/modules/recycling/belt-placer.dm]: Oh, you might be right. I'll limit the number they can store, would 50 be too much?

**Comment by Citinited** [comment]
@Fox-McCloud Yeah, the sprite is very much a placeholder. Problem is that I have no idea what this ovject would look like...

File: code/modules/recycling/belt-placer.dm (hunk 1 above)
Comment (taukausanake) [reply code/modules/recycling/belt-placer.dm]: 50 is how much a plant bag can carry of plant stuff. It should be fine there

**Comment by Citinited** [comment]
I've added better sprites to this (see OP), and limited the number of belts that can be held by the bluespace one.

**Comment by taukausanake** [comment]
Could you make the bluespace one look more like the normal one, but maybe the window be the bluespace animation? I feel it should look more alike the normal one and less like a Rapid Part Dispenser

**Comment by Citinited** [comment]
I changed how the bluespace one was coloured, and added a red light, so hopefully they are sufficiently different now. OP updated

(I'm preserving this picture for comparison - this is how it used to look)
image

File: code/modules/recycling/belt-placer.dm [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+/obj/item/storage/conveyor/attackby(obj/item/conveyor_switch_construct/S, mob/user, params) //So we can link belts en masse
+	if(istype(S))
+		var/linked = FALSE //For nice message
+		for(var/obj/item/conveyor_construct/C in src)
+			C.id = S.id
+			linked = TRUE
+		if(linked)
+			to_chat(user, "<span class='notice'>All belts in [src] linked with [S].</span>")
+	else
+		..()
```
Comment (Crazylemon64) [inline code/modules/recycling/belt-placer.dm]: This should be `return ..()` instead

File: code/modules/recycling/belt-placer.dm [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	name = "bluespace conveyor belt placer"
+	desc = "This device facilitates the rapid deployment of conveyor belts. It utilises bluespace in order to hold many more belts than its regular counterpart."
+	icon_state = "bluespace_belt_placer"
+	item_state = "bluespace_belt_placer"
+	w_class = WEIGHT_CLASS_NORMAL
+	storage_slots = 50
+	max_combined_w_class = 200 //50 belts
+	origin_tech = "engineering=2;bluespace=1"
+
+/obj/item/storage/conveyor/attackby(obj/item/conveyor_switch_construct/S, mob/user, params) //So we can link belts en masse
```
Comment (Crazylemon64) [inline code/modules/recycling/belt-placer.dm]: This should be `obj/item` instead of specifically the conveyor switch.

**Comment by Citinited** [comment]
@Crazylemon64 All issues addressed

**Pull request merged**